keras_nlp/utils/python_utils_test.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_nlp.utils.python_utils import classproperty
+from keras_nlp.utils.python_utils import format_docstring
+
+
+class ClassWithClassProperty:
```

Comment:
Why aren't these objects created within the test fixture `setUp()`? I think it's OK in this case but you can get weird results passing mutable objects to many tests.

Reply:
Hmm, no objects are created here, just class definitions.

I could define a class in setup, attach it to self, and use it in the unit tests, but testing a function/class bound as a property seems like a less pure version of what we are testing here.

Probably the best would be to just keep all functions and class definitions in local scope in a test. I believe they won't get created in that case until the test is run, but otherwise behave similar to here. Will try that.